### PR TITLE
Revoke received/offered Maskinporten delegations API

### PR DESCRIPTION
**Pull request by jonkjetiloye**
## Description
- New controller endpoints for revoking an offered or received MP delegation:
POST `/accessmanagement/api/v1/:party/delegations/maskinportenschema/offered/revoke`
POST `/accessmanagement/api/v1/:party/delegations/maskinportenschema/received/revoke`
- DelegationsService implementation used by both endpoints as only difference is the required authorization for the reportee being either the delegating or receiving party.
- Cleaned up some unneeded mocking of httpContextAccessorMock in DelegationsControllerTest and replaced some with Roles test data between the test user and reportee

## Related Issue(s)
- #179

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.AccessManagement.Core/Services/DelegationsService.cs [hunk 1]
```diff
@@ -224,6 +224,86 @@ public async Task<List<Delegation>> GetMaskinportenSchemaDelegations(string supp
             return await GetAllMaskinportenSchemaDelegations(supplierPartyId, consumerPartyId, scope);
         }
 
+        /// <inheritdoc/>
+        public async Task<DelegationOutput> RevokeMaskinportenDelegation(int authenticatedUserId, DelegationInput delegation)
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Core/Services/DelegationsService.cs:228]: Wondering if we should rename the models used here for both delegation and revoke:
DelegationInput -> DelegationLookup ?
DelegationOutput -> DelegationResult/DelegationActionResult ?

File: src/Altinn.AccessManagement.Core/Services/DelegationsService.cs [hunk 2]
```diff
@@ -224,6 +224,86 @@ public async Task<List<Delegation>> GetMaskinportenSchemaDelegations(string supp
...
+                return output;
+            }
+
+            List<RequestToDelete> policiesToDelete = DelegationHelper.GetRequestToDeleteResourceRegistryService(authenticatedUserId, resourceRegistryId, fromParty.PartyId, toParty.PartyId);
+
+            await _pap.TryDeleteDelegationPolicies(policiesToDelete);
+            return output;
+        }
+
+        private async Task<(DelegationOutput Output, string ResourceRegistryId, Party FromParty, Party ToParty)> ValidateMaskinportenDelegationModel(DelegationInput delegation)
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Core/Services/DelegationsService.cs:242]: The error messages either need to be updated to be correctly referring to revoke and not delegation, or be made generic. to be used in both contexts? e.g.:
"Maskinporten schema delegation must specify..." -> "This operation requires ..."?

File: src/Altinn.AccessManagement.Core/Services/DelegationsService.cs (hunk 1 above)
Comment (jonkjetiloye) [reply src/Altinn.AccessManagement.Core/Services/DelegationsService.cs]: Models has been renamed DelegationLookup and DelegationActionResult

File: src/Altinn.AccessManagement.Core/Services/DelegationsService.cs (hunk 2 above)
Comment (jonkjetiloye) [reply src/Altinn.AccessManagement.Core/Services/DelegationsService.cs]: Both delegation and revoke now use the same validation method.

File: src/Altinn.AccessManagement.Core/Helpers/RightsHelper.cs [hunk 3]
```diff
@@ -50,5 +51,18 @@ public static List<AttributeMatch> GetDelegationSubjectAttributeMatches(Delegati
...
+        /// <summary>
+        /// Builds a RightsQuery request model for lookup of a users rights for a given resource registry service on behalf of the given reportee party
+        /// </summary>
+        public static RightsQuery GetRightsQueryForResourceRegistryService(int userId, string resourceRegistryId, int fromPartyId)
+        {
+            return new RightsQuery
+            {
+                To = new List<AttributeMatch> { new AttributeMatch { Id = AltinnXacmlConstants.MatchAttributeIdentifiers.UserAttribute, Value = userId.ToString() } },
+                From = new List<AttributeMatch> { new AttributeMatch { Id = AltinnXacmlConstants.MatchAttributeIdentifiers.PartyAttribute, Value = fromPartyId.ToString() } },
+                Resource = new AttributeMatch { Id = AltinnXacmlConstants.MatchAttributeIdentifiers.ResourceRegistryAttribute, Value = resourceRegistryId }.SingleToList()
```
Comment (TheTechArch) [inline src/Altinn.AccessManagement.Core/Helpers/RightsHelper.cs:64]: Foreløpig mest en kommentar til meg selv. Hvordan håndteres søk på gitt type

Comment (jonkjetiloye) [reply src/Altinn.AccessManagement.Core/Helpers/RightsHelper.cs:64]: Søk etter alle rettigheter man har for en gitt ResourceType? Er ikke noe vi har støtte for øyeblikket, men det er vel ikke værre enn at man utvider RightsQuery modellen med mulighet for å spesifisere type?

File: test/Altinn.AccessManagement.Tests/Controllers/DelegationsControllerTest.cs [hunk 4]
```diff
@@ -2362,6 +2333,280 @@ public async Task MaskinportenDelegation_ValidationProblemDetails_TooLowAuthenti
...
+            // Assert
+            Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
+        }
+
+        /// <summary>
+        /// Test case: RevokeReceivedMaskinportenScopeDelegation performed by authenticated user 20000095 for the reportee party 50004221 of the jks_audi_etron_gt maskinporten schema resource from the resource registry,
+        ///            which have received delegation from the party 50005545
+        ///            In this case:
+        ///            - The user 20000095 is ADMAI (Tilgangsstyrer) for the To party 50004221
+        /// Expected: RevokeReceivedMaskinportenScopeDelegation returns 204 No Content
```
Comment (TheTechArch) [inline test/Altinn.AccessManagement.Tests/Controllers/DelegationsControllerTest.cs:2553]: Feil excected

Comment (jonkjetiloye) [reply test/Altinn.AccessManagement.Tests/Controllers/DelegationsControllerTest.cs]: Fixed for both this and RevokeReceivedMaskinportenScopeDelegation_ValidationProblemDetails_InvalidFrom case below

**Review by TheTechArch — APPROVED** [review]
💯

**Pull request merged**